functions/openhabian.bash

```diff
@@ -193,86 +193,33 @@ ua-netinst_check() {
   fi
 }
 
-choose_ipv6() {
-  local IPV6_DISABLE="net.ipv6.conf.all.disable_ipv6"
-
-  # shellcheck disable=SC2154
-  if [[ "$ipv6" == "disable" ]]; then
-    if ! grep -qE "^${IPV6_DISABLE}" "${SYSCTL_INIT}"; then
-      disable_ipv6 yes
-    fi
-  else
-    if [[ "$ipv6" == "enable" ]]; then
-      disable_ipv6 no
-    fi
-  fi
-
-  sysctl -p
-}
-
-# arguments: "yes", "no"
-disable_ipv6() {
-  local APT_IPV6_DISABLE='Acquire::ForceIPv4 "true";'
-  local APT_INIT=/etc/apt/apt.conf/S90force-ipv4
-  local IPV6_DISABLE="net.ipv6.conf.all.disable_ipv6"
-  local SYSCTL_INIT="/etc/sysctl.d/99-sysctl.conf"
-
-  if [[ "$1" == "yes" ]]; then
-    echo "${APT_IPV6_DISABLE}" | sudo tee "${APT_INIT}"
-
-    if ! grep -qE "^${IPV6_DISABLE}" "${SYSCTL_INIT}"; then
-      ( echo ""
-        echo "###################################################################"
-        echo "# disable all IPv6 functionality" >> "${SYSCTL_INIT}"
-        echo "${IPV6_DISABLE}=1"
-      )	>> "${SYSCTL_INIT}"
-    else
-      sed -i "s/^# ${IPV6_ENABLE}.*/${IPV6_ENABLE}/g" "${SYSCTL_INIT}"
-    fi
-  else
-    rm -f "${APT_INIT}"
-    sed -i "s/^${IPV6_ENABLE}.*/# ${IPV6_ENABLE}/g" "${SYSCTL_INIT}"
-  fi
-}
-
-choose_ipv6() {
-  local IPV6_DISABLE="net.ipv6.conf.all.disable_ipv6"
-
-  # shellcheck disable=SC2154
-  if [[ "$ipv6" == "disable" ]]; then
-    if ! grep -qE "^${IPV6_DISABLE}" "${SYSCTL_INIT}"; then
-      disable_ipv6 yes
-    fi
-  else
-    if [[ "$ipv6" == "enable" ]]; then
-      disable_ipv6 no
-    fi
-  fi
-
-  sysctl -p
-}
-
-# arguments: "yes", "no"
-disable_ipv6() {
-  local APT_IPV6_DISABLE='Acquire::ForceIPv4 "true";'
-  local APT_INIT=/etc/apt/apt.conf/S90force-ipv4
-  local IPV6_DISABLE="net.ipv6.conf.all.disable_ipv6"
-  local SYSCTL_INIT="/etc/sysctl.d/99-sysctl.conf"
-
-  if [[ "$1" == "yes" ]]; then
-    echo "${APT_IPV6_DISABLE}" | sudo tee "${APT_INIT}"
-
-    if ! grep -qE "^${IPV6_DISABLE}" "${SYSCTL_INIT}"; then
-      ( echo ""
-        echo "###################################################################"
-        echo "# disable all IPv6 functionality" >> "${SYSCTL_INIT}"
-        echo "${IPV6_DISABLE}=1"
-      )	>> "${SYSCTL_INIT}"
-    else
-      sed -i "s/^# ${IPV6_ENABLE}.*/${IPV6_ENABLE}/g" "${SYSCTL_INIT}"
-    fi
-  else
-    rm -f "${APT_INIT}"
-    sed -i "s/^${IPV6_ENABLE}.*/# ${IPV6_ENABLE}/g" "${SYSCTL_INIT}"
-  fi
+## Enable / Disable IPv6 according to the users configured option in '$CONFIGFILE'
+##
+##    config_ipv6()
+##
+config_ipv6() {
+  local aptConf
+  local sysctlConf
+
+  aptConf="/etc/apt/apt.conf/S90force-ipv4"
+  sysctlConf="/etc/sysctl.d/99-sysctl.conf"
+
+  if [[ "${ipv6:-enable}" == "disable" ]]; then
+    echo -n "$(timestamp) [openHABian] Disabling IPv6... "
+    {
+      echo "# Disable all IPv6 functionality"
+      echo "net.ipv6.conf.all.disable_ipv6=1"
+      echo "net.ipv6.conf.default.disable_ipv6=1"
+      echo "net.ipv6.conf.lo.disable_ipv6=1"
+    } >> "$sysctlConf"
```

Comment:
'>>' or '>' ? what if that's executed for the 2nd time ?

maybe not today but if there was a menu option ....

Reply:
I didn't know if there was any other stuff that `/etc/sysctl.d/99-sysctl.conf` was used for so I didn't want to override it. If there is nothing else then I'll just create a includes file and use that.

Reply:
Changed. @mstormi is that better?